### PR TITLE
Terra Experimental Powerpack

### PR DESCRIPTION
**Pull request by tyeagg**
## About The Pull Request
powerpack for energy weapons self recharges and around 4 times ammo cap of normal energy magazine
image
get: better sprites

## Why It's Good For The Game
new niche for energy weapons but needing to sacrifice backpack space

## Changelog
:cl:
add: te-pp10 powerpack
/:cl:

### CODE REVIEW

**Comment by FosterGrime-Heymynameslouie** [comment]
something I didnt even know we needed
very cool pr that finally does something about laser weapons

also I wonder if lumi would make a loadout out of this for hvh

**Review by Kuro020 — CHANGES_REQUESTED** [review]
Sprite needs work, ping me on artwork later so I can destroy your hopes and dreams

**Comment by Lumipharon** [comment]
The charge rate is pretty high. Volkite weapons use notably more charge than lasers.
And related to that, why are you nerfing the laser mg?

**Comment by Losenis** [comment]
image

**Comment by revelation-8468** [comment]
can you link laser guns directly into the power pack?

**Comment by ilikesaying** [comment]
> can you link laser guns directly into the power pack?

Looking at the accepted magazines for the lasguns change, looks like you can.

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Comment by Autisem** [comment]
cool concept but hand chargers are honestly more effective as of right now. especialy when you lose backpack slot for it. maby making energy weapons stronger somehow? otherwise its quiet steep price

**Comment by Losenis** [comment]
PLEASE TELL ME THIS WILL WORK WITH THE POWERFIST

**Comment by Al-1ce** [comment]
It looks like there'll be about five of these for free roundstart. What's stopping someone from taking one off and putting on another for "infinite" ammo?

**Comment by UntoldTactics** [comment]
yes pls
time to be support unga and recharge my comrades cells

**Review by BraveMole — CHANGES_REQUESTED** [review]
Do not change laser balance just to add your power pack

**Comment by Lumipharon** [comment]
That's still a high charge rate, especially consider the capacity buff.
Volkite uses 30 or 36 charge per shot with a charge rate of 32.
Lasers charge is a bit wack, but this is still effectively infinite bullets. 625 shots with 2.5 regenning every 2 seconds? You would to be spamming ridiculously hard to ever really use that up.

**Comment by tyeagg** [comment]
> That's still a high charge rate, especially consider the capacity buff. Volkite uses 30 or 36 charge per shot with a charge rate of 32. Lasers charge is a bit wack, but this is still effectively infinite bullets. 625 shots with 2.5 regenning every 2 seconds? You would have to be spamming ridiculously hard to ever really use that up.

tested constant fire with laser mg and nope it doesnt recharge faster than laser mg

**Comment by Lumipharon** [comment]
> > That's still a high charge rate, especially consider the capacity buff. Volkite uses 30 or 36 charge per shot with a charge rate of 32. Lasers charge is a bit wack, but this is still effectively infinite bullets. 625 shots with 2.5 regenning every 2 seconds? You would have to be spamming ridiculously hard to ever really use that up.
> 
> tested constant fire with laser mg and nope it doesn't recharge faster than laser mg

It doesn't need to is my point. As long as you never run out, how much charge you have left doesn't matter. It just acts as a huge buffer, you're realistically never going to run out unless you're literally running around without ever letting go of mouse 1.

**Review by Pariah919 — COMMENTED** [review]
Seems wack, talk to me on discord.

File: code/modules/projectiles/magazines/energy.dm [hunk 1]
```diff
@@ -160,3 +160,12 @@
...
 		to_chat(user, span_warning("This cell is already at maximum charge!"))
+
+/obj/item/cell/lasgun/volkite/powerpack/energy
+	name = "\improper TE-PK10 powerpack"
+	desc = "A heavy reinforced backpack with an array of ultradensity energy cells, linked to a miniature radioisotope thermoelectric generator for continuous power generation. Used to power Energy Weapons. Click drag cells to the powerpack to recharge."
+	icon = 'icons/obj/items/storage/storage.dmi'
+	icon_state = "laser_powerpack"
+	charge_overlay = "laser_back"
+	maxcharge = 2500
+	charge_amount = 10
```
Comment (TiviPlus) [inline code/modules/projectiles/magazines/energy.dm:171]: I dont get why this has self recharge but rest lgtm

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by helper-2** [comment]
Wait, you want to add a backpack that stores energy worth of 4 lasgun batteris, when even the usual satchel stores 5 of them, and that powerbank don't selfrecharge itself.
Do I understand correctly?

**Comment by helper-2** [comment]
I don't want this pr to die.

**Comment by FosterGrime-Heymynameslouie** [comment]
it was never meant to be

**Pull request closed without merging**